main.tf

```diff
@@ -28,7 +28,13 @@ resource "google_healthcare_dicom_store" "dicom_stores" {
     for s in var.dicom_stores :
     s.name => s
   }
-  name    = each.value.name
+  name = each.value.name
+  dynamic "notification_config" {
+    for_each = lookup(each.value, "pubsub_topic", "") != "" ? [each.value.pubsub_topic] : []
```

Comment:
In general I try to avoid making changes that don't match the underlying API. The reason is there could be extra fields added down the line under notification_config (e.g. maybe something to configure what gets sent or how frequently or other outputs). Should we keep notification_config as a field?

Reply:
Note that CFT in general *does* try to provide simplifications on top of the underlying API where it makes sense. Either way is valid, so it's mostly about which tradeoff you want to favor.

Reply:
Changed to use notification_config to make future extension easier.